### ISSUE TITLE
Average CD results

### ISSUE THREAD

**Issue by narthnil** [AnTao97/UnsupervisedPointCloudFeatureLearning#3]
Hi,

thanks for this great repo. I was wondering with which method you calculated the average CD score reported in the README.md file (I could not find a function to do so). When I was calculating the CD score for one of your pretrained models (Original, 16, Sphere), I do get a much higher CD score (~21).

Thanks,

Linh

**Comment by antao97** [comment]
The CD loss is calculated in `loss.py`. You need to run `main.py` with `--eval` and report the CD loss in test set.

**Comment by narthnil** [comment]
Hi,
thanks for your reply.
I also did use the --eval function, but I get an even higher score. I downloaded your model as a way to check whether I did something wrong and ran the --eval function. This is what I get:

```
python main.py --eval --model_path snapshot/Reconstruct_foldnet_sphere/models/shapenetcorev2_278.pkl --dataset_root data --encoder foldnet --k 16 --shape sphere --dataset shapenetcorev2 --batch_size 16 --gpu 2
Namespace(batch_size=16, dataset='shapenetcorev2', dataset_root='data', dropout=0.5, encoder='foldnet', epochs=None, eval=True, exp_name=None, feat_dims=512, gpu='2', k=16, model_path='snapshot/Reconstruct_foldnet_sphere/models/shapenetcorev2_278.pkl', no_cuda=False, num_points=2048, shape='sphere', snapshot_interval=10, task='reconstruct', use_jitter=False, use_rotate=False, use_translate=False, workers=16)
Inference set size (train): 35708
Inference set size (test): 10261
Load model from snapshot/Reconstruct_foldnet_sphere/models/shapenetcorev2_278.pkl
Train set 0 for SVM saved.
Train set 1 for SVM saved.
Train set 2 for SVM saved.
Train set 3 for SVM saved.
Train set 4 for SVM saved.
Train set 5 for SVM saved.
Train set 6 for SVM saved.
Train set 7 for SVM saved.
Train set 8 for SVM saved.
Train set 9 for SVM saved.
Train set 10 for SVM saved.
Train set 11 for SVM saved.
Train set 12 for SVM saved.
Train set 13 for SVM saved.
Train set 14 for SVM saved.
Train set 15 for SVM saved.
Train set 16 for SVM saved.
Train set 17 for SVM saved.
Avg loss 62.937129974365234
Finish generating train set for SVM.
Test set 0 for SVM saved.
Test set 1 for SVM saved.
Test set 2 for SVM saved.
Test set 3 for SVM saved.
Test set 4 for SVM saved.
Test set 5 for SVM saved.
Avg loss 65.18447875976562
Finish generating test set for SVM.
Loading feature dataset...
Training set size: 35708
Testing set size: 10261
Transfer linear SVM accuracy: 81.90%
```

The loss sums over the whole batch, and the average loss averages over the list of batches. So this represents the average batch loss. When dividing this by batch size 16 (to get a per datapoint cd) I still get 3.9335 (train) and 4.0740 (test), which is still not too close to what you get.

Best,

Linh

**Comment by antao97** [comment]
When you evaluate the CD loss, you need to change the CD loss in this repo into the CD loss in FoldingNet paper:
![截屏2021-10-11 上午10 24 22](https://user-images.githubusercontent.com/41320807/136724605-ff016db8-40db-4bb4-ae84-5ca0b3b4b133.png)

Because each shape has 2048 points. The CD loss in FoldingNet can be roughly calculated by 65.18447875976562 / 16 / 2048 / 2 * 10^4 = 9.946362115442753
